pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnitTestShouldIncludeAssertRule.java

```diff
@@ -31,16 +36,29 @@ public UnitTestShouldIncludeAssertRule() {
 
     @Override
     public Object visit(ASTMethodDeclaration method, Object data) {
-        ASTBlock body = method.getBody();
+        boolean usesSoftAssertExtension = usesSoftAssertExtension(method.getEnclosingType());
         Set<String> extraAsserts = getProperty(EXTRA_ASSERT_METHOD_NAMES);
+        Predicate<ASTMethodCall> isAssertCall = usesSoftAssertExtension
+                ? TestFrameworksUtil::isSoftAssert
+                : TestFrameworksUtil::isProbableAssertCall;
```

Comment:
this assumes they are exclusive, and I wonder if this is accurate… maybe we should replace this to something such as…

```suggestion
        Predicate<ASTMethodCall> isAssertCall = TestFrameworksUtil::isProbableAssertCall;
        if (usesSoftAssertExtension) {
            isAssertCall = isAssertCall.or(TestFrameworksUtil::isSoftAssert);
        }
```

Reply:
I think I understand your point: One could use the extension to use softassertions, but still use normal assertions.

After reading https://assertj.github.io/doc/#assertj-core-soft-assertions I figured, I misunderstood the usage of [assertSoftly](https://assertj.github.io/doc/#assertj-core-assertSoftly-assertions) - it is an alternative to the JUnit5 extension and implicitly also calls assertAll at the end.

I'll update the PR soon.